### PR TITLE
s/pull/dev/ GCS staging for pull.*gke

### LINKED ISSUE #6584: GKE pull tests are broken because release is in an unexpected gcs path
```
W0201 16:52:17.074] Creating cluster e2e-139...
W0201 16:52:32.197] ...............done.
W0201 16:52:33.110] ERROR: (gcloud.container.clusters.create) Operation [<Operation
W0201 16:52:33.110]  endTime: u'2018-02-01T16:52:31.318769411Z'
W0201 16:52:33.111]  name: u'operation-1517503937051-ab9bf336'
W0201 16:52:33.111]  operationType: OperationTypeValueValuesEnum(CREATE_CLUSTER, 1)
W0201 16:52:33.111]  selfLink: u'https://test-container.sandbox.googleapis.com/v1/projects/943316867058/zones/us-central1-f/operations/operation-1517503937051-ab9bf336'
W0201 16:52:33.111]  startTime: u'2018-02-01T16:52:17.051260112Z'
W0201 16:52:33.112]  status: StatusValueValuesEnum(DONE, 3)
W0201 16:52:33.112]  statusMessage: u'cluster_api_version not found.'
W0201 16:52:33.112]  targetLink: u'https://test-container.sandbox.googleapis.com/v1/projects/943316867058/zones/us-central1-f/clusters/e2e-139'
W0201 16:52:33.112]  zone: u'us-central1-f'>] finished with error: cluster_api_version not found.
```

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes #6584

/assign @krzyzacy @mikedanese

### CODE REVIEW

File: jobs/config.json [hunk 1]
```diff
@@ -10945,7 +10945,7 @@
       "--gke-environment=test",
       "--gke-shape={\"default\":{\"Nodes\":4,\"MachineType\":\"n1-standard-2\"}}",
       "--provider=gke",
-      "--stage=gs://kubernetes-release-pull/ci/pull-kubernetes-e2e-gke",
+      "--stage=gs://kubernetes-release-dev/ci/pull-kubernetes-e2e-gke",
```
Comment (krzyzacy) [inline jobs/config.json]:
```
--stage=gs://kubernetes-release-dev/ci
--stage-suffix=pull-gke-gci
```
I think
/shrug

Comment (BenTheElder) [reply jobs/config.json]: done

File: jobs/config.json [hunk 2]
```diff
@@ -11311,4 +11313,4 @@
       "sig-big-data"
     ]
   }
-}
+}
```
Comment (krzyzacy) [inline jobs/config.json]: nit for a new line in the end?

**Comment by krzyzacy** [comment]
looks like you also need to bring back the config_test change

**Comment by mikedanese** [comment]
what will allow this to merge? it's blocking some PRs.

**Comment by BenTheElder** [comment]
@mikedanese I need to patch one of the presubmits but I've been busy working on a bigger bug. will fix soon.

**Comment by BenTheElder** [comment]
Also for *reasons* we really need to support other buckets..

**Comment by BenTheElder** [comment]
@mikedanese this should go in now :crossed_fingers:

**Pull request merged**